scenarioo-server/src/main/java/org/scenarioo/dao/search/elasticsearch/ElasticSearchAdapter.java

```diff
@@ -42,156 +43,183 @@
 import org.scenarioo.rest.base.BuildIdentifier;
 import org.scenarioo.rest.search.SearchRequest;
 
+import java.io.IOException;
 import java.net.InetAddress;
-import java.net.UnknownHostException;
 import java.util.ArrayList;
 import java.util.List;
 
 public class ElasticSearchAdapter implements SearchAdapter {
 
-    private final static Logger LOGGER = Logger.getLogger(ElasticSearchAdapter.class);
+	private final static Logger LOGGER = Logger.getLogger(ElasticSearchAdapter.class);
 
 	// It's ok that this is static because the Elasticsearch endpoint config can not be changed
 	// at runtime but only by restarting Scenarioo. If we ever make it possible to change
 	// the endpoint config in the UI (i.e. at runtime) then the client can not be created only
 	// once anymore but has to be recreated when changing the config.
-	private static TransportClient client;
+	private static RestHighLevelClient restClient;
+	private static boolean elasticSearchConfigured = true;
 
 	private final ConfigurationRepository configurationRepository = RepositoryLocator.INSTANCE
 		.getConfigurationRepository();
 
 	private final String endpoint = configurationRepository.getConfiguration().getElasticSearchEndpoint();
 
-	private final String clusterName = configurationRepository.getConfiguration().getElasticSearchClusterName();
-
-    public ElasticSearchAdapter() {
-		if (client != null) {
-			// already initialized
+	public ElasticSearchAdapter() {
+		if (restClient != null || !elasticSearchConfigured) {
+			// already initialized or no configuration provided.
 			return;
 		}
 
 		if (!isSearchEndpointConfigured()) {
 			LOGGER.info("no valid elasticsearch endpoint configured.");
+			elasticSearchConfigured = false;
 			return;
 		}
 
 		try {
 			int portSeparator = endpoint.lastIndexOf(':');
 			String host = endpoint.substring(0, portSeparator);
 			int port = Integer.parseInt(endpoint.substring(portSeparator + 1), 10);
-			client = new PreBuiltTransportClient(Settings.builder()
-				.put("cluster.name", clusterName).build())
-				.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
-		} catch (UnknownHostException e) {
-			LOGGER.warn("No elasticsearch cluster running.");
+			restClient = new RestHighLevelClient(
+					RestClient.builder(new HttpHost(InetAddress.getByName(host), port, "http")));
+			//Sometimes a deadlock occurs when a CreateIndexRequest is created, by accessing Settings.Builder.EMPTY_SETTINGS first this can be stopped.
+			//noinspection unused
+			Settings emptySettings = Settings.Builder.EMPTY_SETTINGS;
```

Comment:
put this three lines in a private method and call it `prepareEmptySettingToAvoidClassloadingDeadlockInElasticSearch`